src/node_realm.cc

```diff
@@ -359,4 +304,88 @@ void Realm::VerifyNoStrongBaseObjects() {
   });
 }
 
+v8::Local<v8::Context> Realm::context() const {
+  return PersistentToLocal::Strong(context_);
+}
+
+#define V(PropertyName, TypeName)                                              \
+  v8::Local<TypeName> PrincipalRealm::PropertyName() const {                   \
+    return PersistentToLocal::Strong(PropertyName##_);                         \
+  }                                                                            \
+  void PrincipalRealm::set_##PropertyName(v8::Local<TypeName> value) {         \
+    PropertyName##_.Reset(isolate(), value);                                   \
+  }
+PER_REALM_STRONG_PERSISTENT_VALUES(V)
+#undef V
+
+PrincipalRealm::PrincipalRealm(Environment* env,
+                               v8::Local<v8::Context> context,
+                               const RealmSerializeInfo* realm_info)
+    : Realm(env, context, kPrincipal) {
+  // Create properties if not deserializing from snapshot.
+  // Or the properties are deserialized with DeserializeProperties() when the
+  // env drained the deserialize requests.
+  if (realm_info == nullptr) {
+    CreateProperties();
+  }
+}
+
+void PrincipalRealm::MemoryInfo(MemoryTracker* tracker) const {
+  Realm::MemoryInfo(tracker);
+}
+
+MaybeLocal<Value> PrincipalRealm::BootstrapRealm() {
+  EscapableHandleScope scope(isolate_);
+
+  MaybeLocal<Value> result = ExecuteBootstrapper("internal/bootstrap/node");
+
+  if (result.IsEmpty()) {
+    return MaybeLocal<Value>();
+  }
+
+  if (!env_->no_browser_globals()) {
+    result = ExecuteBootstrapper("internal/bootstrap/web/exposed-wildcard");
+
+    if (result.IsEmpty()) {
+      return MaybeLocal<Value>();
+    }
+
+    result = ExecuteBootstrapper("internal/bootstrap/web/exposed-window");
+
+    if (result.IsEmpty()) {
+      return MaybeLocal<Value>();
+    }
```

Comment:
I think this can be simplified as

```suggestion
    if (ExecuteBootstrapper("internal/bootstrap/web/exposed-wildcard").IsEmpty() ||
        ExecuteBootstrapper("internal/bootstrap/web/exposed-window").IsEmpty()) {
      return MaybeLocal<Value>();
    }
```

This is already a thing in existing code, but this part looks particularly repetitive probably because of the module id..

Reply:
Done.